cmd/get_ids.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+)
+
+// GetCommitIDFromArgs returns the first CLI argument and exits gracefully if it's missing
+func GetCommitIDFromArgs() string {
+	// Note: os.Args[] always contains at least one value (the path to the program).
+	// Any additional values that exist are the arguments passed to the program.
+	if len(os.Args) == 1 {
```

Comment:
Is there a reason this can't be encoded as a function input parameter rather than as a side effect?

e.g. applied to `GetHexIDFromCommitID(GetCommitIDFromArgs(os.Args)`

Reply:
Good idea. I'll work this into my next commit after merging. 👍